docs/pages/result.rst

```diff
@@ -43,6 +43,41 @@ Make sure to check out how to compose container with
 ``flow`` or :ref:`pipe`!
 Read more about them if you want to compose your containers easily.
 
+Pattern Matching
+----------------
+
+``Result`` values can be matched using the new feature of Python 3.10,
+`Structural Pattern Matching <https://www.python.org/dev/peps/pep-0622/>`_,
+see the example below:
+
+.. code:: python
+
+  from returns.result import Success, Failure, safe
+
+  @safe
+  def div(first_number: int, second_number: int) -> int:
+      return first_number // second_number
+
+  match div(1, 0):
+      # Matches if the result stored inside `Success` is `10`
+      case Success(10):
+          print('Result is "10"')
+      # Same as above but using match-by-name
+      case Success(inner_value=20):
+          print('Result is "20"')
+      # Matches any `Success` instance and bind its value to the ``value`` variable
+      case Success(value):
+          print('Result is "{}"'.format(value))
+      # Matches any `Success` instance
```

Comment:
I think that this case can be removed, we already have `catch-all` case above. And having two is an anti-pattern.

Reply:
It should be a great `rule` for WPS to check, shouldn't it?